tests/e2e/hci.go

```diff
@@ -488,4 +495,336 @@ var _ bool = ginkgo.Describe("hci", func() {
+		2.  Create statefulset with 1 replica.
+		3.  Isolate a host which has a k8s-worker with attached PVs in cluster1.
+		4.  Wait for 5 mins, verify that the k8s-worker is restarted and brought up on another host.
+		5.  Verify that the replicas on the k8s-worker come up successfully.
+		6.  Scale up statefulset to 3 replicas.
+		7.  Re-integrate the host used in step 3
+		8.  Scale up statefulset to 5 replicas and verify they are successful
+		9.  Cleanup all objects created during the test
+	*/
+	ginkgo.It("HCI mesh - One host isolation", ginkgo.Label(p0, block, vanilla, hci), func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var err error
+		var replicas int32 = 1
+		hostIp := ""
+
+		scParameters = map[string]string{}
+		scParameters[scParamStoragePolicyName] = remoteStoragePolicyName
+		scSpec := getVSphereStorageClassSpec(defaultNginxStorageClassName, scParameters, nil, "", "", false)
+		sc, err := client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Create a sts with 1 replica")
+		sts := createCustomisedStatefulSets(ctx, client, namespace, false, replicas,
+			false, nil, false, true, "", "", sc, "")
+		defer func() {
+			ginkgo.By(fmt.Sprintf("Deleting all statefulsets in namespace: %v", namespace))
+			fss.DeleteAllStatefulSets(ctx, client, namespace)
+		}()
+
+		nodes, err := client.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		pods := fss.GetPodList(ctx, client, sts)
+		targetWorkerName := pods.Items[0].Spec.NodeName
+
+		ginkgo.By("Isolate the host which has a k8s-worker with attached PVs in cluster1")
+		workerNode, err := client.CoreV1().Nodes().Get(ctx, targetWorkerName, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		targetHost := e2eVSphere.getHostFromVMReference(ctx, getHostMoref4K8sNode(
+			ctx, client, workerNode))
+		finder := find.NewFinder(e2eVSphere.Client.Client, false)
+		var datacenters []string
+		cfg, err := getConfig()
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		dcList := strings.Split(cfg.Global.Datacenters,
+			",")
+		for _, dc := range dcList {
+			dcName := strings.TrimSpace(dc)
+			if dcName != "" {
+				datacenters = append(datacenters, dcName)
+			}
+		}
+
+		for _, dc := range datacenters {
+			defDc, err := finder.Datacenter(ctx, dc)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			finder.SetDatacenter(defDc)
+			hosts, err := finder.HostSystemList(ctx, "*")
+			for _, host := range hosts {
+				if host.Reference().Reference().Value == targetHost.Value {
+					hostInfo := host.Common.InventoryPath
+					hostIpInfo := strings.Split(hostInfo, "/")
+					hostIp = hostIpInfo[len(hostIpInfo)-1]
+					break
```

Comment:
can you put this in a helper function.

Reply:
Will do it in next PR

Reply:
This is just used in this testcase. In the next testcase I need the IP of other hosts in the cluster so that I can make network partition